### ISSUE TITLE
Compromised?

### ISSUE THREAD

**Issue by DarekDan** [warrenday/graphql-network-inspector#164]
**Describe the bug**
This extension has been reported as compromised in the global supply chain attack. is this still a case?

https://www.securityweek.com/cyberhaven-chrome-extension-hack-linked-to-widening-supply-chain-campaign/

**Comment by DarekDan** [comment]
More details: https://www.bleepingcomputer.com/news/security/new-details-reveal-how-hackers-hijacked-35-google-chrome-extensions/

**Comment by DarekDan** [comment]
And more: https://secureannex.com/blog/cyberhaven-extension-compromise/

**Comment by warrenday** [comment]
Hey @DarekDan 

Thanks for the heads up here. 

The extension was patched, version 2.22.6 was effected and 2.22.7+ is secure. 

Appreciate you raising the issue. I was made aware of the attack quickly and took immediate action to resolve the issue.

**Comment by DarekDan** [comment]
Appreciate the quick turnaround.

**Comment by indera** [comment]
Was there a code change to fix the issue? Which commit was it?

I am asking because the tool was banned and we need a proof of the fix. Thanks for the great work!

**Comment by warrenday** [comment]
@indera the code itself on github was not changed, a malicious oauth application on our developer account pushed a change directly to the chrome store, circumventing github. The patch simply rolled back this push. Removed the oauth app, and further secured the developer account to increase detection of malicious oauth apps.

**Comment by jasisk** [comment]
Any chance we can get some more info such as timestamps of when the compromised package was live? Does the extension store give you a sense of number of folks who updated to the malicious version?

**Comment by warrenday** [comment]
The malicious code was trying to capture ChatGPT cookies, the malicious version 2.22.6 was released by the attacker on 30th December during the morning. I was contacted by a cyber security agency that same day informing me of the release. I removed the malicious app and rolled back the version at around 1pm which changed the version to 2.22.7. We are now on 2.22.9 at the time of writing.

The developer dashboard does not give me exact times, but I have been speaking to a few cyber security agencies who had been monitoring a number of extensions, which I suspect is why they were able to detect and contact me so quickly.

The recommendation I gave at the time was to click "log out of all devices" on ChatGPT.

To be susceptible to the attack you would need 2.22.6 installed and have visited ChatGPT during the morning and early afternoon of December 30th (UTC timezone).

The current number of users with 2.22.6 installed is zero, however on 30th December it was 5,361. Roughly 5% of the user base.

<img width="784" alt="Screenshot 2025-01-05 at 08 53 55" src="https://github.com/user-attachments/assets/de7e8b0c-5798-448e-bcef-4b13699ee8c2" />
